core/src/main/java/org/elasticsearch/index/mapper/DocumentParser.java

```diff
@@ -888,21 +873,14 @@ private static void parseCopy(String field, ParseContext context) throws IOExcep
                     mapper = context.docMapper().objectMappers().get(context.path().pathAsText(paths[i]));
                     if (mapper == null) {
                         // One mapping is missing, check if we are allowed to create a dynamic one.
-                        ObjectMapper.Dynamic dynamic = parent.dynamic();
-                        if (dynamic == null) {
-                            dynamic = dynamicOrDefault(context.root().dynamic());
-                        }
+                        ObjectMapper.Dynamic dynamic = dynamicOrDefault(parent, context.root().dynamic());
 
                         switch (dynamic) {
                             case STRICT:
                                 throw new StrictDynamicMappingException(parent.fullPath(), paths[i]);
                             case TRUE:
                                 Mapper.Builder builder = context.root().findTemplateBuilder(context, paths[i], "object");
                                 if (builder == null) {
-                                    // if this is a non root object, then explicitly set the dynamic behavior if set
-                                    if (!(parent instanceof RootObjectMapper) && parent.dynamic() != ObjectMapper.Defaults.DYNAMIC) {
-                                        ((ObjectMapper.Builder) builder).dynamic(parent.dynamic());
-                                    }
```

Comment:
Hrm, while I agree this is different, I don't think what I have here is incorrect, as far as the behavior of other dynamic fields.  The code here and in parseObject didn't always work (would be broken for propagating dynamic setting when a dynamic template exists). And this particular line in parseCopy didn't work at all, because it would be executing when `builder == null`.  I think we should separately look at how to fix dynamic propagation (which I agree is buggy, but before this change).